### DISCUSSION TITLE
Safari 14.1 Support?

### DISCUSSION THREAD

**Discussion by antonpug** [JamesBrill/react-speech-recognition#94] [Q&A]
I just downloaded the Safari 14.1 Beta which supports Web Speech API, the demo of react-speech-recognition shows the Start/Stop/Reset buttons, but no microphone access request occurs, nor any recording happens. 

@JamesBrill do you know if this is something that should work natively in react-speech-recognition and is likely a bug with Safari, or does something need to be done to the package to make it work with Safari?

**Comment by JamesBrill** [comment]
Hi @antonpug thanks for the heads-up! If Safari is going to start supporting speech recognition natively, that's an exciting development. I imagine `react-speech-recognition` just needs to be updated to recognise Apple's native SpeechRecognition object. I'll have a play and see if I can get them working together.

**Reply by JamesBrill** [reply]
I just downloaded the Safari 14.1 Beta on MacOS and ran the `react-speech-recognition` test harness in there. It worked fine, activating the microphone and transcribing in different languages.

I did have to go through a couple of popups to give permission for Safari to perform the speech recognition. They looked a bit like this:
<img width="366" alt="Screenshot 2021-05-21 at 13 11 30" src="https://user-images.githubusercontent.com/2140027/119135411-42ffe900-ba36-11eb-8c97-39db0946cdfe.png">
<img width="554" alt="Screenshot 2021-05-21 at 13 13 12" src="https://user-images.githubusercontent.com/2140027/119135447-514e0500-ba36-11eb-8205-fb3acf68a02f.png">

After that, it works exactly as it does on Chrome:

<img width="204" alt="Screenshot 2021-05-21 at 13 16 42" src="https://user-images.githubusercontent.com/2140027/119135822-d6391e80-ba36-11eb-9c93-b7ce51c4950f.png">

From the Safari 14.1 release notes: `Note that users will need Siri enabled in System Preferences on macOS or Settings in iOS or iPadOS for the API to be available to be used.`

Do you have Siri enabled? And which platform do you use?

**Reply by antonpug** [reply]
Thank you! I didn't have Siri enabled, so that was the culprit. Still not seeing it show up in X Code simulator Safari but that could still need an update from their side.

**Reply by jkolko** [reply]
Hi James, is there a way to check for Siri enabled, so I can hide the buttons if clicking them will do nothing? Thanks -
